### DISCUSSION TITLE
Problem detecting host in my several subnetworks

### DISCUSSION THREAD

**Discussion by fgonzalezmacias** [netalertx/NetAlertX#868] [Q&A]
Hello

I'm testing NetAlertx in my organization and with the ARP-SCAN only obtain hosts in the same subnet of the docker server (10.84.6.0/24), but in the other several subnets I don't detect any host.

I have tested with NMAP plugin but unsucessfully. I don't obtain any host in the scan process. Since the docker server with NMAP I can detect any host in other subnets.

![image](https://github.com/user-attachments/assets/e46f9843-9f39-411a-80ce-b1f6eb38bb79)

My configuration of the NMAP plugin is the next one:

![image](https://github.com/user-attachments/assets/99cc10fc-4019-4a66-b212-c79733b67a07)

And I can see the log with the result of the nmap process, but none devices are added in NetAlert. I only have the devices of the same subnet of the docker server.

![image](https://github.com/user-attachments/assets/531714f7-1294-4005-8ff3-6a652a4eef6c)

Any advice?

Thanks in advance.

**Comment by jokob-sk** [comment]
Hi @fgonzalezmacias,

Did you read the following documentation explaining the subnet accessibility of various scanning methods? https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md

The Nmap you have enabled is for ports, look at NMAPDEV to try to discover devices.

**Comment by fgonzalezmacias** [comment]
Yes, I read this page, but I don't find direct information about detect new devices and NMAPDEV. Yes in plugins web page. Thank yout
But I follow with the same problem. I configure several subnetworks and detect many hosts, but it aren't added to NetAlertX

![image](https://github.com/user-attachments/assets/f7181d90-c7dd-4f9b-b141-3c9cfb3a4677)

![image](https://github.com/user-attachments/assets/45b83c7e-038b-4880-8804-6da75d56cf57)

I only have 88 devices that are detected by arp-scan, but no new devices by nmapdev

![image](https://github.com/user-attachments/assets/19c164c2-d175-48cc-8b1d-1f36c0e24528)

Any idea?

Thanks in advance

**Reply by jokob-sk** [reply]
Hi there,

Please check if you have a mac associated with the devices discovered by the NMAPDEV plugin. MACs are used as a unique ID and if not supplied (not being able to detect it) those entries will be skipped. You can check the discovered plugin entries under Integrations -> Plugins -> NMAPDEV

If a MAC address is not present, you can follow the linked guide:

> If direct scans are not possible, you can use [supplementing plugins](https://github.com/jokob-sk/NetAlertX/blob/main/front/plugins/README.md) that use alternate methods. Protocols used by the SNMPDSC or DHCPLSS plugins have good support and usually can be used as a workaround.
